src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1176,6 +1190,107 @@ class FunctionBinaryArithmetic : public IFunction
         return column_to;
     }
 
+    ColumnPtr executeDateTime64Subtraction(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type,
+                                           size_t input_row_count) const
+    {
+        using OpImpl = DecimalBinaryOperation<Op, Decimal64, /* check_overflow */ true>;
```

Comment:
I think it's supposed to be controlled by a setting: https://github.com/ClickHouse/ClickHouse/blob/3f5006095f481f1716bccc4481ac7977ab7c5b0e/src/Functions/FunctionBinaryArithmetic.h#L1481
https://github.com/ClickHouse/ClickHouse/blob/6152f52ef04ed8553cb69d9cbe19d070009aa9e6/src/Core/Settings.cpp#L2941

Reply:
Thanks, addressed the concern by calling `helperInvokeEither` which checks `decimal_check_overflow`